### PR TITLE
Check database availability instead of trying to create

### PR DESCRIPTION
**Pull request by ptman**
https://github.com/wallabag/docker/issues/190

### CODE REVIEW

**Comment by j0k3r** [comment]
Shouldn't the setup fail if the ping is false?
Looks like it pass everything (or tests aren't accurate about that).

**Comment by ptman** [comment]
You're right. Weird module that it doesn't fail if it cannot reach. I also looked at ansible-lint output

**Comment by ptman** [comment]
The logs are a bit inpenetrable. Any idea what fails?

File: root/etc/ansible/entrypoint.yml [hunk 1]
```diff
@@ -111,37 +111,16 @@
...
-        login_host="{{ database_host }}"
-        port={{ database_port }}
-        login_user="{{ database_root_user_postgres }}"
-        login_password="{{ database_root_password_postgres }}"
+    - name: verify PostgreSQL server is available
+      postgresql_ping:
+        login_host: '{{ database_host}}'
+        login_user: '{{ database_user }}'
+        login_password: '{{ database_password }}'
+        db: '{{ database_name }}'
```
Comment (Spiritus44) [inline root/etc/ansible/entrypoint.yml:119]: I think you forget to specify the port of the pgsql server.
Consider using `port: '{{ database_port }}'`

**Review by Spiritus44 — CHANGES_REQUESTED** [review]
Hi,
Thanks for your PR.

First i'm not sure to understand why you delete the "posgresql db" and "add postgresql user" tasks. I my mind you should just add the "verify PostgreSQL server is available" task before theses 2 tasks ?

Next you forget to specify the port in your "verify PostgreSQL server is available" task. it's a problem for people like me who use custom port for their pgsql database.

To finish, I tryed to add your task in front of the other pgsql task but I have this error in the log : `[localhost] TASK [verify PostgreSQL server is available] bin dev entrypoint.sh etc home lib media mnt opt proc root run sbin srv sys tmp usr var fatal: [localhost]: FAILED! => {"msg": "The conditional check 'not pgping.available' failed. The error was: error while evaluating conditional (not pgping.available): 'dict object' has no attribute 'available'"}`
Did I missed something ?

**Comment by ptman** [comment]
> First i'm not sure to understand why you delete the "posgresql db" and "add postgresql user" tasks

Creating the database shouldn't be the concern of the wallabag image/container, but instead handled by the postgres container using the provided POSTGRES_DB, POSTGRES_USER and POSTGRES_PASSWORD environment variables ( https://hub.docker.com/_/postgres )

**Pull request closed without merging**